include/seqan3/io/sequence/sequence_file_format_fasta.hpp

```diff
@@ -94,15 +100,130 @@ namespace seqan3
+              typename seq_type,        // other constraints checked inside function
+              typename id_type,
+              typename qual_type,
+              typename seq_qual_type>
+    void write(stream_type                     & stream,
+               sequence_file_out_options const & options,
+               seq_type                       && sequence,
+               id_type                        && id,
+               qual_type                      && SEQAN3_DOXYGEN_ONLY(qualities),
+               seq_qual_type                  && seq_qual)
+    {
+        static_assert(detail::decays_to_ignore_v<seq_type> || detail::decays_to_ignore_v<seq_qual_type>,
+                      "Either the sequence field, or the seq_qual field need to be set to std::ignore.");
+        static_assert(detail::decays_to_ignore_v<qual_type> || detail::decays_to_ignore_v<seq_qual_type>,
+                      "Either the qualities field, or the seq_qual field need to be set to std::ignore.");
+
+        ranges::ostreambuf_iterator stream_it{stream};
+
+        // ID
+        if constexpr (detail::decays_to_ignore_v<id_type>)
+        {
+            throw std::logic_error{"The ID field may not be set to ignore when writing FASTA files."};
+        }
+        else
+        {
+            if (ranges::empty(id)) //[[unlikely]]
+                throw std::runtime_error{"The ID field may not be empty when writing FASTA files."};
+
+            write_id(stream_it, options, id);
+        }
+
+        // Sequence
+        if constexpr (!detail::decays_to_ignore_v<seq_type>) // sequence
+        {
+            if (ranges::empty(sequence)) //[[unlikely]]
+                throw std::runtime_error{"The SEQ field may not be empty when writing FASTA files."};
+
+            write_seq(stream_it, options, sequence);
+        }
+        else if constexpr (!detail::decays_to_ignore_v<seq_qual_type>) // seq_qual
+        {
+            if (ranges::empty(seq_qual)) //[[unlikely]]
+                throw std::runtime_error{"The SEQ_QUAL field may not be empty when writing FASTA files."};
+
+            write_seq(stream_it, options, seq_qual);
+        }
+        else
+        {
+            throw std::logic_error{"The SEQ and SEQ_QUAL fields may not both be set to ignore when writing FASTA files."};
+        }
+    }
+
 protected:
     //!\privatesection
     //!\brief Implementation of reading the ID.
     template <typename stream_view_t,
-              typename options_type,
+              typename seq_legal_alph_type,
               typename id_type>
-    void read_id(stream_view_t & stream_view,
-                 options_type  & options,
-                 id_type       & id)
+    void read_id(stream_view_t                                       & stream_view,
+                 sequence_file_in_options<seq_legal_alph_type> const & options,
+                 id_type                                             & id)
```

Comment:
I found an error:
Lines 247 and 248 have to be swapped: for ID you first skip `>` and then read the line

Reply:
It doesn't make a difference either way, does it? (For 237f it does that's why its the other way around)